src/mi2.ts

```diff
@@ -171,36 +171,40 @@ export class MI2 extends EventEmitter implements IDebugger {
-                        this.log('stdio','TTY: sleep ' + sleepVal + ';');  
-                        // wls - const wsl_process = ChildProcess.exec("cmd.exe /c start bash -c 'sleep "+sleepVal+"'");                      
-                        if (isWslSsh)
-                            this.createTerminal("vscode", sleepVal, target);
-                        else 
-                            this.createTerminal(gdbttytype, sleepVal, target);
-                        const sleep = async (milliseconds) => {
-                            await new Promise(resolve => setTimeout(resolve, milliseconds));
-                        }
-                        let try_find=0;
-                        while(try_find<4){
-                                await sleep(500);
-                                xterm_device = this.findTtyName(target);
-                                try_find++;
-                                if(xterm_device!="") break;
+                let device_ext_display_os = "";
+                if(gdbtty){               
+                    if (process.platform !== "win32"){
+                        let xterm_device = this.findTtyName(target);
+                        const isWslSsh = vscode.env.remoteName === "wsl" || vscode.env.remoteName === "ssh-remote" ;
+                        if(xterm_device === ""){
+                            let sleepVal=this.hashCode(target);
+                            this.log('stdio','TTY: sleep ' + sleepVal + ';');  
+                            // wls - const wsl_process = ChildProcess.exec("cmd.exe /c start bash -c 'sleep "+sleepVal+"'");                      
+                            if (isWslSsh)
+                                this.createTerminal("vscode", sleepVal, target);
+                            else 
+                                this.createTerminal(gdbtty, sleepVal, target);
+                            const sleep = async (milliseconds) => {
+                                await new Promise(resolve => setTimeout(resolve, milliseconds));
+                            }
+                            let try_find=0;
+                            while(try_find<4){
+                                    await sleep(500);
+                                    xterm_device = this.findTtyName(target);
+                                    try_find++;
+                                    if(xterm_device!="") break;
+                            }
+                            if(xterm_device==="") this.log("stderr","tty: Install 'xterm' to use gdb's tty option\n");
                         }
-                        if(xterm_device==="") this.log("stderr","tty: Install 'xterm' to use gdb's tty option\n");
+                        if(xterm_device.includes("pts")){
+                            this.gdbArgs.push("--tty="+xterm_device );
+                            device_ext_display_os = "linux";
+                        }        
+                    }else if(process.platform === "win32"){
```

Comment:
this elseif is the direct inversion of the previous if - an else should be enough, no?

Reply:
Well noticed! I will correct.

Reply:
OK! I will correct. Thanks.